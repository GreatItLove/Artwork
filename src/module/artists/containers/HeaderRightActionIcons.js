import { connect } from 'react-redux';

import HeaderRightActionIcons from '../../common/components/header/HeaderRightActionIcons';
import { openArtistAddDialog, artistDownloadPathRequest, openArtistDownloadDialog } from '../artistsActions';

const mapDispatchToProps = dispatch => ({
  openAddDialog: () => {
    dispatch(openArtistAddDialog());
  },
  openDownloadDialog: () => {
    dispatch(artistDownloadPathRequest());
    dispatch(openArtistDownloadDialog());
  },
});

export default connect(null, mapDispatchToProps)(HeaderRightActionIcons);
