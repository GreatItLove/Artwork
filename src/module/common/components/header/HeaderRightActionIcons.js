import React from 'react';

import DownloadIcon from './DownloadIcon';
import PrintIcon from './PrintIcon';
import AddIcon from './AddIcon';

/**
 * @function HeaderRightActionIcons
 * @returns {XML}
 */
function HeaderRightActionIcons({ openAddDialog, openDownloadDialog }) {
  return (
    <div>
      <DownloadIcon openDownloadDialog={openDownloadDialog} />
      <PrintIcon />
      <AddIcon openAddDialog={openAddDialog} />
    </div>
  );
}

export default HeaderRightActionIcons;
