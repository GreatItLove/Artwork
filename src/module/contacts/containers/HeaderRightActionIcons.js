import { connect } from 'react-redux';

import HeaderRightActionIcons from '../../common/components/header/HeaderRightActionIcons';
import { openContactAddDialog, contactDownloadPathRequest, openContactDownloadDialog } from '../contactActions';

const mapDispatchToProps = dispatch => ({
  openAddDialog: () => {
    dispatch(openContactAddDialog());
  },
  openDownloadDialog: () => {
    dispatch(contactDownloadPathRequest());
    dispatch(openContactDownloadDialog());
  },
});

export default connect(null, mapDispatchToProps)(HeaderRightActionIcons);
