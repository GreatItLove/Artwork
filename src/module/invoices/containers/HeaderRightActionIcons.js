import { connect } from 'react-redux';

import HeaderRightActionIcons from '../../common/components/header/HeaderRightActionIcons';
import { invoiceDownloadPathRequest, openInvoiceDownloadDialog } from '../invoiceActions';

const mapDispatchToProps = dispatch => ({
  openDownloadDialog: () => {
    dispatch(invoiceDownloadPathRequest());
    dispatch(openInvoiceDownloadDialog());
  },
});

export default connect(null, mapDispatchToProps)(HeaderRightActionIcons);
