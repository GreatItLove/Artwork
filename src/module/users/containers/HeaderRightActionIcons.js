import { connect } from 'react-redux';

import HeaderRightActionIcons from '../../common/components/header/HeaderRightActionIcons';
import { openUserAddDialog, usersDownloadPathRequest, openUserDownloadDialog } from '../usersActions';

const mapDispatchToProps = dispatch => ({
  openAddDialog: () => {
    dispatch(openUserAddDialog());
  },
  openDownloadDialog: () => {
    dispatch(usersDownloadPathRequest());
    dispatch(openUserDownloadDialog());
  },
});

export default connect(null, mapDispatchToProps)(HeaderRightActionIcons);
