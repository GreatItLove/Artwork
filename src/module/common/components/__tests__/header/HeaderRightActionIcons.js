import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import HeaderRightActionIcons from '../../header/HeaderRightActionIcons';

describe('HeaderRightActionIcons', () => {
  it('Snapshot of HeaderRightActionIcons', () => {
    const wrapper = shallow(<HeaderRightActionIcons />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
