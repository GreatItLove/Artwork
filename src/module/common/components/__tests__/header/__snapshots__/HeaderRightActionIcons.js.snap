// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRightActionIcons Snapshot of HeaderRightActionIcons 1`] = `
<div>
  <MuiComponent />
  <MuiComponent />
  <MuiComponent />
</div>
`;
